internals/daemon/daemon.go

```diff
@@ -747,9 +747,34 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 	return nil
 }
 
-// stopRunningServices stops all running services, waiting for a short time
-// for them all to stop.
+// stopRunningServices waits for all running changes to complete then stops all running services,
+// waiting for a short time for them all to stop.
 func (d *Daemon) stopRunningServices() error {
+	st := d.state
+	st.Lock()
+	var runningChanges []*state.Change
```

Comment:
It seems to me we should restrict this to kind="start" (starting services) and status=DoingStatus, no? And probably call it `startChanges`.